toolkits/search/arcade_search/tools/google_shopping.py

```diff
@@ -0,0 +1,66 @@
+from typing import Annotated, Any, Optional
+
+from arcade.sdk import ToolContext, tool
+from arcade.sdk.errors import ToolExecutionError
+
+from arcade_search.constants import (
+    DEFAULT_GOOGLE_SHOPPING_COUNTRY,
+    DEFAULT_GOOGLE_SHOPPING_LANGUAGE,
+)
+from arcade_search.google_data import GOOGLE_DOMAIN_BY_COUNTRY_CODE
+from arcade_search.utils import (
+    call_serpapi,
+    extract_shopping_results,
+    prepare_params,
+    resolve_country_code,
+    resolve_language_code,
+)
+
+
+@tool(requires_secrets=["SERP_API_KEY"])
+async def search_shopping_products(
+    context: ToolContext,
+    keywords: Annotated[
+        str,
+        "Keywords to search for products in Google Shopping. E.g. 'Apple iPhone'.",
+    ],
+    country_code: Annotated[
+        Optional[str],
+        "2-character country code to search for products in Google Shopping. "
+        f"E.g. 'us' (United States). Defaults to '{DEFAULT_GOOGLE_SHOPPING_COUNTRY or 'us'}'.",
+    ] = DEFAULT_GOOGLE_SHOPPING_COUNTRY,
+    language_code: Annotated[
+        Optional[str],
+        "2-character language code to search for news articles. E.g. 'en' (English). "
+        f"Defaults to '{DEFAULT_GOOGLE_SHOPPING_LANGUAGE or 'en'}'.",
+    ] = DEFAULT_GOOGLE_SHOPPING_LANGUAGE,
+) -> Annotated[dict[str, list[dict[str, Any]]], "News results."]:
+    """Search for news articles related to a given query."""
```

Comment:
update docstring - looks like it still has the contents from a news search tool

Reply:
fixed